cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -501,33 +506,171 @@ __inline__ __device__ std::pair<char const*, char const*> trim_whitespaces_quote
- * @param end Pointer to the first element after the string
- * @param prefix String we're searching for at the start of the input range
+ * @param[in] begin Beginning of the character string
+ * @param[in] end End of the character string
+ * @param opts The global parsing behavior options
+ *
+ * @return The parsed numeric value
  */
-template <int N>
-__inline__ __device__ auto skip_if_starts_with(char const* begin,
-                                               char const* end,
-                                               const char (&prefix)[N])
+template <typename T, int base>
+__inline__ __device__ T decode_value(char const* begin,
+                                     char const* end,
+                                     parse_options_view const& opts)
 {
-  static constexpr size_t prefix_len = N - 1;
-  if (end - begin < prefix_len) return begin;
-  return thrust::equal(thrust::seq, begin, begin + prefix_len, prefix) ? begin + prefix_len : begin;
+  return cudf::io::parse_numeric<T, base>(begin, end, opts);
 }
 
 /**
- * @brief Finds the first element after the leading space characters.
+ * @brief Decodes a numeric value base on templated cudf type T
  *
- * @param begin Pointer to the first element of the string
- * @param end Pointer to the first element after the string
+ * @param[in] begin Beginning of the character string
+ * @param[in] end End of the character string
+ * @param opts The global parsing behavior options
+ *
+ * @return The parsed numeric value
  */
-__inline__ __device__ auto skip_spaces(char const* begin, char const* end)
+template <typename T, CUDF_ENABLE_IF(!cudf::is_timestamp<T>() and !cudf::is_duration<T>())>
+__inline__ __device__ T decode_value(char const* begin,
+                                     char const* end,
+                                     parse_options_view const& opts)
+{
+  return cudf::io::parse_numeric<T>(begin, end, opts);
+}
+
+template <typename T, CUDF_ENABLE_IF(cudf::is_timestamp<T>())>
+__inline__ __device__ T decode_value(char const* begin,
+                                     char const* end,
+                                     parse_options_view const& opts)
 {
-  return thrust::find_if(thrust::seq, begin, end, [](auto elem) { return elem != ' '; });
+  return to_timestamp<T>(begin, end, opts.dayfirst);
 }
 
+template <typename T, CUDF_ENABLE_IF(cudf::is_duration<T>())>
+__inline__ __device__ T decode_value(char const* begin, char const* end, parse_options_view const&)
+{
+  return to_duration<T>(begin, end);
+}
+
+struct ConvertFunctor {
+  /**
+   * @brief Dispatch for numeric types whose values can be convertible to
+   * 0 or 1 to represent boolean false/true, based upon checking against a
+   * true/false values list.
+   *
+   * @return bool Whether the parsed value is valid.
+   */
+  template <typename T,
+            std::enable_if_t<std::is_integral_v<T> and !std::is_same_v<T, bool> and
+                             !cudf::is_fixed_point<T>()>* = nullptr>
```

Comment:
Missed one -- the one I originally commented on. 😆
```suggestion
            CUDF_ENABLE_IF(std::is_integral_v<T> and !std::is_same_v<T, bool> and !cudf::is_fixed_point<T>())>
```

Reply:
Embarrassing. Sorry. Fixed it now 👍